src/index.js

```diff
@@ -99,90 +102,99 @@ const yargsv = require("yargs")(process.argv.slice(2))
+            duration: DEFAULT_DURATION,
+            externalId: DEFAULT_EXTERNAL_ID,
+            mfaToken: DEFAULT_MFA_TOKEN,
+            mfaTokenArn: DEFAULT_MFA_TOKEN_ARN,
+            sessionName: DEFAULT_SESSION_NAME,
+            verbose: DEFAULT_VERBOSE_VALUE
     });
-}
-catch (error) {
-    switch (error.errorType) {
-        case ERROR_CONFLICTING_ROLE_ARN_AND_PROFILE:
-            console.log('Only one of a role arn or a profile can be specified');
-            break;
-        case ERROR_INCOMPLETE_MFA_OPTIONS:
-            console.error(`To use MFA you must supply both --mfa-token-arn and --mfa-token. Missing value: ${error.errorDetail}`);
-            break;
-        case ERROR_INVALID_ROLE_ARN:
-            console.log(`Invalid role arn provided. Provided value: ${error.errorDetail}`);
-            break;
-        case ERROR_MISSING_ROLE_ARN_AND_PROFILE:
-            console.log('Either a role arn or a profile must be specified');
-            break;
-        default:
-            console.log('An unknown error occurred.', error.message);
-            break;
-    }
+    const positionalOptions = objectEntriesFilter(extractPositionalOptions(yargsv.command), { roleArn: NO_ROLE_ARN });
+    const profileOptionsValues = yargsv.profile !== NO_PROFILE ? getProfileOptionsValues(yargsv.profile) : {};
 
-    process.exit(1);
-}
-
-if (options.verbose) {
-    if (options.roleArn !== NO_ROLE_ARN) {
-        console.log(`Using RoleArn: ${options.roleArn}`);
-    }
-    else {
-        console.log(`Using Profile: ${options.profile}`);
+    try {
+        options = new Options({
+            ...profileOptionsValues,
+            ...specifiedOptions,
+            ...positionalOptions
+        });
+
+        if (options.verbose) {
+            console.log('options', options);
+        }
     }
-}
-
-(async () => {
-    let command;
-    let credentials;
-    const stsOptions = {};
+    catch (error) {
+        switch (error.errorType) {
+            case ERROR_CONFLICTING_ROLE_ARN_AND_PROFILE:
+                console.log('Only one of a role arn or a profile can be specified');
+                break;
+            case ERROR_INCOMPLETE_MFA_OPTIONS:
+                console.error(`To use MFA you must supply both --mfa-token-arn and --mfa-token. Missing value: ${error.errorDetail}`);
+                break;
+            case ERROR_INVALID_ROLE_ARN:
+                console.log(`Invalid role arn provided. Provided value: ${error.errorDetail}`);
+                break;
+            case ERROR_MISSING_ROLE_ARN_AND_PROFILE:
+                console.log('Either a role arn or a profile must be specified');
+                break;
+            default:
+                console.log('An unknown error occurred.', error.message);
+                break;
+        }
 
-    try {
-        const assumeRoleParameters = {
-            RoleSessionName: options.sessionName,
-            DurationSeconds: options.duration
-        };
+        process.exit(1);
```

Comment:
I went ahead and differentiated the exit codes as much as possible in ecd6f61 
IMO this really begs for documentation to identify what those codes are. However I'm inclined to not delay this change for documentation, because it's functionally equivalent and a proper effort to produce a man page (or similar) could be a PR in its own right.

Thoughts?